scripts/check-binary.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+BASE_URL="https://binaries.soliditylang.org/bin"
+REPO_ROOT="$(dirname "$0")/.."
+LIST_FILE=/tmp/list.json
+
+function fail() {
+    echo -e "ERROR: $@" >&2
+    exit 1
+}
+
+function check_release_version() {
+    current_version=$1
+
+    curl --silent --fail "$BASE_URL/list.json" -o $LIST_FILE
+    [[ ! -f $LIST_FILE ]] && fail "Download of release list failed:\n    [url]: $BASE_URL/list.json"
```

Comment:
I see you removed the curly braces everywhere :) That's not really what I meant. I only said they are kinda superfluous if the variable stands on its own. I do agree that it's a good practice to have them when it's a part of a string though:
```suggestion
    [[ ! -f $LIST_FILE ]] && fail "Download of release list failed:\n    [url]: ${BASE_URL}/list.json"
```

Reply:
By the way, maybe we should add something about shell style in [`CODING_STYLE.md`](https://github.com/ethereum/solidity/blob/develop/CODING_STYLE.md)?

Unless we can configure shellcheck to catch these automatically. We should at least have shellcheck in solc-js too.

Reply:
Yes, it would be great indeed. Are there any of the default rules usually ignored by you? Or should we stick with their general recommendations?
We can add shellcheck as a pre-commit hook for `.sh` files (https://github.com/koalaman/shellcheck/blob/master/README.md#pre-commit).

Reply:
Their general recommendations are very good. I can actually only think of two I'd disable globally:
- [SC2016: Expressions don't expand in single quotes, use double quotes for that](https://www.shellcheck.net/wiki/SC2016): This is useless because it has tons of false-positives. It basically does not let you use single quotes inside double quotes ever, even in valid cases. And there's no "fix", you have to ignore the warning.
- [SC2002: Useless cat. Consider `cmd < file | ..` or `cmd file | ..` instead](https://www.shellcheck.net/wiki/SC2002): I know it's theoretically less efficient but in practice this does not matter and I find the `cat file | cmd1 | cmd2` form much more logical and readable. I prefer it over something like `cmd1 < file | cmd2` and `< file cmd1 | cmd2` just looks weird.